### PR TITLE
allow consumers to control what enrichments are fetched

### PR DESCRIPTION
**Pull request by tomwayson**
The reason for this PR is to allow consumers to control what enrichments are fetched via a `requestOptions.enrichments` property as proposed in #538.

It also refactors how we enrich content by exposing new  function`enrichContent(content, requestOptions)` which calls a new `fetchContent(content, requestOptions)`.  `enrichContent()` is called by `getContentFromPortal()` and `getContentFromHub()` (instead of at the end of `getContent()`). This means that you can expect those functions to include the same enrichments as `getContent()`. This also means that a lot of 🧀 has moved out of the content module and corresponding test and into a new enrichments module and test. 

All in all I'm much happier w/ the refactored functions and new APIs and they should make it easier to implement and test #322.

### CODE REVIEW

File: packages/content/src/enrichments.ts [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+enum DatePrecision {
+  Year = "year",
+  Month = "month",
+  Day = "day",
+  Time = "time"
+}
+
+function getMetadataPath(identifier: keyof IMetadataPaths) {
+  // NOTE: i have verified that this will work regardless of the "Metadata Style" set on the org
+  const metadataPaths: IMetadataPaths = {
```
Comment (rweber-esri) [inline packages/content/src/enrichments.ts:60]: Could probably just define this as a `const` outside the scope of this `fn` in place of `IMetadataPaths`?

File: packages/content/src/enrichments.ts [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+    date = new Date(+parts[0], +parts[1] - 1, +parts[2]);
+    precision = DatePrecision.Day;
+  } else if (!Number.isNaN(Date.parse(isoString))) {
+    // any other string parsable to a valid date
+    date = new Date(isoString);
+    precision = isoString.includes("T")
+      ? DatePrecision.Time
+      : DatePrecision.Day;
+  }
+  return date && precision && { date, precision };
```
Comment (rweber-esri) [inline packages/content/src/enrichments.ts:116]: Maybe deserving of an interface?

File: packages/content/src/enrichments.ts [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+ * and b) you cannot naively call new Date with an ISO 8601 string that does not include time information
+ * For example, when I, here in mountain time, do new Date('2018').getFullYear() I get "2017".
+ * This is because when you do not provide time or timezone info, UTC is assumed, so new Date('2018') is 2018-01-01T00:00:00 in UTC
+ * which is actually 7 hours earlier here in mountain time.
+ *
+ * @param {string} isoString
+ * @return { date: Date, precision: DatePrecision }
+ */
+export function parseISODateString(isoString: string) {
+  isoString = `${isoString}`;
```
Comment (rweber-esri) [inline packages/content/src/enrichments.ts:92]: For some reason I _thought_ our lint rules complained about argument reassignment.This is probably fine, but might be better practice to not mutate/reassign args.

File: packages/content/src/enrichments.ts [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+};
+
+const fetchMetadata = (
+  content: IHubContent,
+  requestOptions?: IHubRequestOptions
+) => getContentMetadata(content.id, requestOptions);
+
+const fetchOwnerOrgId = (
```
Comment (rweber-esri) [inline packages/content/src/enrichments.ts:245]: can we expose this? I need for [PR-9667](https://github.com/ArcGIS/opendata-ui/pull/9667/files#diff-076d4969300d0bb9d112746b1dfc71f68ff2e0abe1ecde160cd68e21885f4f6fR258) unless changes in this PR will result in `orgId` being present on content created from `composer` datasets.

File: packages/content/src/enrichments.ts [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+  return (
+    content.type === "Web Map" &&
+    contentTypeKeywords.some(
+      typeKeyword => hubTypeKeywords.indexOf(typeKeyword) > -1
+    )
+  );
+};
+
+const shouldFetchOrgId = (content: IHubContent) => {
+  return !content.orgId && isHubCreatedContent(content);
```
Comment (rweber-esri) [inline packages/content/src/enrichments.ts:299]: Maybe we don't need to expose `fetchOwnerOrgId` if this handles detecting content coming from composer?

**Review by rweber-esri — APPROVED** [review]
Looks good, only minor comments.

**Review by mjuniper — COMMENTED** [review]
Love it, nice work! Thanks, Tom!

File: packages/content/src/enrichments.ts (hunk 3 above)
Comment (tomwayson) [reply packages/content/src/enrichments.ts:92]: Yeah, seems to me our rules got a little looser a while back, I assume when we added prettier

File: packages/content/src/enrichments.ts (hunk 4 above)
Comment (tomwayson) [reply packages/content/src/enrichments.ts:245]: so, I swear I left a comment about this last night, but it seems to have gotten lost.

I definitely can, but you can also achieve the same w/ `const { orgId } = fetchEnrichments(content, { ...requestOptions, enrichments: ['orgId'] }`

That is a little awkward, so if you think it's better just to expose each function, I can do that too.

File: packages/content/src/enrichments.ts (hunk 5 above)
Comment (tomwayson) [reply packages/content/src/enrichments.ts:299]: this does not detect content coming from composer.

**Pull request merged**